### PR TITLE
Set reactive callbacks based on whether or not cursor is ordered

### LINKED ISSUE #3: Finds inside server autoruns use polling instead of oplog
Originally from comment(s): https://github.com/peerlibrary/meteor-reactive-publish/issues/11#issuecomment-418943191

### Expected functionality
When using a `find` (or `findOne`, or anything which triggers a `fetch`) within a [server-side autorun](https://github.com/peerlibrary/meteor-server-autorun), I would expect that the find uses the oplog observe driver rather than polling.

For example, if I have:

```
Meteor.publish('messagesForGroup', function (groupId) {
  this.autorun(() => {
    const group = Groups.findOne(groupId, { fields: { members: 1 } });
    if (group.members.indexOf(this.userId) > -1) {
      return Messages.find({ group: group._id });
    }
  });
});
```

The expectation would be that the `Groups.findOne(groupId, { fields: { members: 1 } });` uses oplog.

### Actual functionality
The [reactive-mongo package adds](https://github.com/peerlibrary/meteor-reactive-mongo/blob/master/server.coffee#L49) `addedBefore` and `movedBefore` callbacks to `fetch`, `map` and `forEach` on cursors.

Because of this, it fails the first check in `observeChanges` in Meteor's [mongo_driver.js](https://github.com/meteor/meteor/blob/d8042c8fa9dd0cc53604b2729ac33333a2ce5b73/packages/mongo/mongo_driver.js#L1291) for whether or not it can use oplog.

The presence of `addedBefore` and `movedBefore` means that `ordered` returns true when the driver calls [LocalCollection._observeChangesCallbacksAreOrdered](https://github.com/meteor/meteor/blob/24865b28a0689de8b4949fb69ea1f95da647cd7a/packages/minimongo/local_collection.js#L1361).

Therefore it is **never** possible in the current state to use oplog on cursors called from within a reactive context. They will always use polling.

### PR DESCRIPTION
**Pull request by etyp**
Fixes #3

### CODE REVIEW

**Comment by etyp** [comment]
@mitar I'm not sure why, but the same [Minimongo test fails every time](https://github.com/meteor/meteor/blob/devel/packages/minimongo/minimongo_tests_client.js#L2219), even when I run the package with no changes (literally cloned your package on my local in a new directory to be sure).

I also tried checking out my local `meteor` directory to [release-1.6](https://github.com/meteor/meteor/tree/release-1.6) which was the version last time I can see CI tests passing in your repo and had no luck.

Finally, I tried forking and submitting a [PR with the only change being an added character in the README](https://github.com/etyp/meteor-reactive-mongo/pull/1) but the test still failed: https://travis-ci.com/etyp/meteor-reactive-mongo

Any ideas? Would you be willing to try on your local to see?

**Comment by mitar** [comment]
I will merge this, but I will do it differently. I think it is better if we just allow users to specify this through options.

**Comment by mitar** [comment]
BTW, do not make pull requests from `master` branch. Always make a special branch and make changes there. If it happens that upstream does not merge your PR, you will get dirty master branch forever. Also, you will have problems making multiple separate PRs.

**Comment by mitar** [comment]
The reason why tests are failing is because now they are run on MongoDB 3.6 which changed behavior how arrays are sorted. See https://docs.mongodb.com/manual/release-notes/3.6-compatibility/#array-sort-behavior. I opened: https://github.com/meteor/meteor/issues/10212.

**Comment by mitar** [comment]
So my plan here is that we do not care about order unless user specifies `sort` option to `find` query, or sets `ordered` to `true` through options (our own extension to options). User should be also able to set `ordered` to `false` to explicitly disable it even if there is `sort`. So having `sort` sets the default value for `ordered` option.

Can you maybe update this PR accordingly, and README?

File: README.md [hunk 1]
```diff
@@ -23,6 +23,37 @@ Installation
...
+```
+
+Alternatively, if you want to force polling while using a `sort` and `limit` (since [depending on your use case, it could be more efficient](https://blog.meteor.com/tuning-meteor-mongo-livedata-for-scalability-13fe9deb8908)), you can add the disableOplog option to the find:
+
+```
+// Server code
+
+// Will use polling
+Tracker.autorun(() => {
+  Collection.find({ topic: 'Programming' }, { sort: { title: 1 }, limit: 50, disableOplog: true });
```
Comment (mitar) [inline README.md]: This example is bad, this does not do anything. Just making a cursor does not do anything. You have to `fetch` documents or something similar.

**Comment by etyp** [comment]
@mitar got it. I've updated the logic accordingly - can you confirm it looks right?

Now it works so that:
- If an `ordered` option is explicitly passed, we'll always use that
- If no `ordered` option is passed, we only use ordered callbacks if a `sort` is present

Not positive if this is what you're going for, so please confirm and I can update the readme once confirmed. I'm trying to avoid continuing to dirty my master branch so keeping commits minimal. If it's easier for you to merge and fix what I've written then let's do that.

**Comment by mitar** [comment]
You understood correctly.

I think in README we should split the discussion about oplog and non-oplog and ordered and non-ordered. I think we first should explain that there is a concept of ordered cursors and that keeping track of ordering means that computation will be invalidated also when order changes. And depending what user wants, can use `ordered` option.

Then, another section should say that users should think about which queries can use oplog and which cannot (as you should in general think) and that there are some general conditions, and also that it does interact with `ordered`.

**Comment by etyp** [comment]
@mitar updated the readme to include a section about ordered/unordered. Let me know if you have any additional suggestions.

I'm hoping we can get a version bump soon as this will make a significant difference in our production app's performance. In a quick test (loading our app and leaving it open for 1 minute), I found that with these changes I could reduce our queries/min to **40%** of what they currently are.

**Comment by mitar** [comment]
Thanks. Released as 0.3.0.

**Pull request merged**